processor/transformprocessor/internal/metrics/functions_test.go

```diff
@@ -25,6 +25,33 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
 )
 
+func Test_newFunctionCall_invalid(t *testing.T) {
+	tests := []struct {
+		name string
+		inv  common.Invocation
+	}{
+		{
+			name: "invalid aggregation temporality",
+			inv: common.Invocation{
+				Function: "convert_gauge_to_sum",
+				Arguments: []common.Value{
+					{
+						String: strp("invalid_agg_temp"),
+					},
+					{
+						Bool: boolp(true),
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := common.NewFunctionCall(tt.inv, DefaultFunctions(), ParsePath)
+			assert.Error(t, err)
+		})
+	}
+}
 func Test_newFunctionCall_NumberDataPoint(t *testing.T) {
```

Comment:
Can you add a "happy path" test case that tests the conversions work successfully?

Reply:
I added these to separate test functions (`Test_newFunctionCall_Metric_Gauge` & `Test_newFunctionCall_Metric_Sum`) because they don't really fit the mold of the `Test_newFunctionCall_NumberDataPoint` test table (e.g. metric must exist on the transform context and I want to test for both sums and gauges).

Reply:
I'm just blind and missed them lol

Reply:
They look good.  Function testing is getting an overall as part of #10101 so I'm good with this implementation for the initial function addtion